nomad/structs/funcs.go

```diff
@@ -531,6 +531,12 @@ func ParsePortRanges(spec string) ([]uint64, error) {
 				return nil, fmt.Errorf("invalid range: starting value (%v) less than ending (%v) value", end, start)
 			}
 
+			// Full range validation is below but prevent creating
+			// arbitrarily large arrays here
+			if end > MaxValidPort {
+				return nil, fmt.Errorf("port must be < %d but found %s", MaxValidPort, part)
```

Comment:
```suggestion
				return nil, fmt.Errorf("port must be < %d but found %s", MaxValidPort, end)
```

Reply:
Do we need to check `start`? Maybe not because `end < start`  is already checked above? 🤔

Reply:
Yeah exactly, only end needs to be checked. Just need to make sure the upper bound is limited to MaxValidPort because the loop below would OOM if it was allowed to be math.MaxUint64 (18446744073709551615!).